master/custom/pr_testing.py

```diff
@@ -111,5 +175,143 @@ def handle_pull_request(self, payload, event):
+            )
+            return (changes, "git")
+
+        pull_url = payload["issue"]["pull_request"]["url"]
+
+        # We need to fetch the PR data from GitHub because the payload doesn't
+        # contain a lot of information we need.
+        pull_request = yield self._get_pull_request(pull_url)
+        if pull_request is None:
+            log.msg("Failed to fetch PR #{} from {}".format(number, pull_url))
+            return (changes, "git")
+
+        repo_full_name = payload["repository"]["full_name"]
+        head_sha = pull_request["head"]["sha"]
+
+        log.msg("Processing GitHub PR #{}".format(number), logLevel=logging.DEBUG)
+
+        head_msg = yield self._get_commit_msg(repo_full_name, head_sha)
+        if self._has_skip(head_msg):
+            log.msg(
+                "GitHub PR #{}, Ignoring: "
+                "head commit message contains skip pattern".format(number)
+            )
+            return ([], "git")
+
+        builder_filter_fn = re.compile(builder_filter)
+        yield self._post_comment(
+            payload["issue"]["comments_url"],
+            BUILD_COMMAND_SCHEDULED_MESSAGE.format(
+                user=payload["sender"]["login"],
+                commit=head_sha,
+                filter=builder_filter,
+                builders="\n".join(
+                    {
+                        f"- `{builder}`"
+                        for builder in self.builder_names
+                        if builder_filter_fn.match(builder)
+                    }
+                ),
+            ),
+        )
+
+        return self._get_changes_from_pull_request(
+            changes, number, payload, pull_request, event, builder_filter
+        )
+
+    @defer.inlineCallbacks
+    def handle_pull_request(self, payload, event):
+
+        changes = []
+        number = payload["number"]
+        action = payload.get("action")
+
+        if action != "labeled":
+            log.msg("GitHub PR #{} {}, ignoring".format(number, action))
+            return (changes, "git")
+
+        label = payload.get("label")["name"]
+        if label not in {TESTING_LABEL, REFLEAK_TESTING_LABEL}:
+            log.msg("Invalid label in PR #{}, ignoring".format(number))
+            return (changes, "git")
+
+        repo_full_name = payload["repository"]["full_name"]
+        head_sha = payload["pull_request"]["head"]["sha"]
+
+        log.msg("Processing GitHub PR #{}".format(number), logLevel=logging.DEBUG)
+
+        head_msg = yield self._get_commit_msg(repo_full_name, head_sha)
+        if self._has_skip(head_msg):
+            log.msg(
+                "GitHub PR #{}, Ignoring: "
+                "head commit message contains skip pattern".format(number)
+            )
+            return ([], "git")
+
+        yield self._remove_label_and_comment(payload, label)
+
+        builder_filter = ""
+        if label == TESTING_LABEL:
+            builder_filter = ".*"
```

Comment:
Should we try to filter out the refleak builders here?

Reply:
Not really, as the current test label means "test with all buildbots" and I don't want people that has been used to that meaning to realize too late that the label didn't ran the refleak ones.